### ISSUE TITLE
Compiler failure on ARM64 Windows

### ISSUE THREAD

**Issue by targos** [simdutf/simdutf#407]
In the (private) Node.js release CI, we're seeing failures with the nightly builds, which contain simdutf 5.2.3 (after https://github.com/nodejs/node/pull/52381).

Here's the error:

```
08:02:09 c:\ws\deps\simdutf\simdutf.cpp(16719,7): error C2664: '__n128x4 neon_ld4m_q8(const char *)': cannot convert argument 1 from 'const uint8_t [64]' to 'const char *' [c:\ws\deps\simdutf\simdutf.vcxproj]
08:02:09 c:\ws\deps\simdutf\simdutf.cpp(16719,7): message : Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast [c:\ws\deps\simdutf\simdutf.vcxproj]
08:02:09 C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\arm64_neon.h(6146,10): message : see declaration of 'neon_ld4m_q8' [c:\ws\deps\simdutf\simdutf.vcxproj]
08:02:09 c:\ws\deps\simdutf\simdutf.cpp(16719,7): message : while trying to match the argument list '(const uint8_t [64])' [c:\ws\deps\simdutf\simdutf.vcxproj]
08:02:09 c:\ws\deps\simdutf\simdutf.cpp(16719,73): fatal  error C1903: unable to recover from previous error(s); stopping compilation [c:\ws\deps\simdutf\simdutf.vcxproj]
```

This happens with MSVC 14.37.32822.

I'm not a Windows specialist myself but I'm happy to provide more information if I can.

**Comment by lemire** [comment]
It says that it fails on the last line:

```C++

  constexpr static uint8_t source_table[64] = {
      'A', 'Q', 'g', 'w', 'B', 'R', 'h', 'x', 'C', 'S', 'i', 'y', 'D',
      'T', 'j', 'z', 'E', 'U', 'k', '0', 'F', 'V', 'l', '1', 'G', 'W',
      'm', '2', 'H', 'X', 'n', '3', 'I', 'Y', 'o', '4', 'J', 'Z', 'p',
      '5', 'K', 'a', 'q', '6', 'L', 'b', 'r', '7', 'M', 'c', 's', '8',
      'N', 'd', 't', '9', 'O', 'e', 'u', '+', 'P', 'f', 'v', '/',
  };
  constexpr static uint8_t source_table_url[64] = {
      'A', 'Q', 'g', 'w', 'B', 'R', 'h', 'x', 'C', 'S', 'i', 'y', 'D',
      'T', 'j', 'z', 'E', 'U', 'k', '0', 'F', 'V', 'l', '1', 'G', 'W',
      'm', '2', 'H', 'X', 'n', '3', 'I', 'Y', 'o', '4', 'J', 'Z', 'p',
      '5', 'K', 'a', 'q', '6', 'L', 'b', 'r', '7', 'M', 'c', 's', '8',
      'N', 'd', 't', '9', 'O', 'e', 'u', '-', 'P', 'f', 'v', '_',
  };
  const uint8x16_t v3f = vdupq_n_u8(0x3f);
  const uint8x16x4_t table =
      vld4q_u8((options & base64_url) ? source_table_url : source_table);
```

The function signature for `vld4q_u8` is `vld4q_u8(uint8_t const * ptr)`.  So this looks correct to me...

https://developer.arm.com/architectures/instruction-sets/intrinsics/vld4q_u8

Let me spin up my ARM-based Windows box.

**Comment by lemire** [comment]
The next release will contain a fix.

**Comment by lemire** [comment]
The issue has been fixed in 5.2.4. I suspect it is a bug in Visual Studio, but we have a workaround.

**Comment by lemire** [comment]
@targos I bought a Windows-based ARM machine *specifically* for this purpose. It is kind of nice and it is interesting to be able to run Windows on ARM.

(Note: I also like x64, I like all processors equally.)
